File: two pointers/lc11.py
def maxArea(height):
    h = len(height)
    if h==2:
          return min(height[0],height[1])
     
    lp = 0
    rp = h-1
    maxA = 0

    while lp<rp:
        area = min(height[lp],height[rp])*(rp-lp)
        maxA = max(maxA, area)
        if height[lp]<height[rp]:
            lp+=1
        else:
            rp-=1
    return maxA

result = maxArea([1,8,6,2,5,4,8,3,7])
print (result)

File: phase 1/01 - two pointers/lc11.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    
    let h = height.length
    
    if (h==2) {return Math.min(height[0],height[1])}

    let lp = 0
    let rp = h-1
    let max = 0

    while (lp<rp){
        let area = Math.min(height[rp],height[lp])*(rp-lp)

        max = Math.max(max,area)
        if (height[lp]<height[rp]){
            lp++
        } else{
            rp--
        }

    }

    return max

};